microsoft/PowerToys
# [Powertoys Run] Implemented "hidden files" flag in native Windows Search

## Summary of the Pull Request
This PR implements the "Hidden Files" flag in Native  Windows Search instead of custom code.

## References

## PR Checklist
* [x] Applies to #5068 (I messed up #5069 which can't be reopened. This PR has a cleaner history as well :) )

## Detailed Description of the Pull Request / Additional comments
The Windows Search API will now handle the "Hidden Files" Flag by using the new generated SQL Query.

## Validation Steps Performed

- I've created the following folder in my users download folder (with a lot of images hidden/ not hidden)  with a similar name, and i ensured modification date for 1 hidden and 1 shown file was most recent 
![powerlauncherindownloadsfolder](https://user-images.githubusercontent.com/16852398/87824286-43a77c00-c875-11ea-8c62-9ea09894cd9c.png)
- I search for the text "powerlauncher".
- I expect my search results to only contain "_shown" postfix files.

I repeat the above steps when i manually put "DisplayHiddenFiles" to true and expect there also be "_hidden" files.

- There's also a unit test to ensure the correct WHERE query is generated / the unit tests are updated.
